src/hooks/useVideoTrackDimensions/useVideoTrackDimensions.tsx

```diff
@@ -10,7 +10,10 @@ export default function useVideoTrackDimensions(track?: TrackType) {
     setDimensions(track?.dimensions);
 
     if (track) {
-      const handleDimensionsChanged = (track: TrackType) => setDimensions(track?.dimensions);
+      const handleDimensionsChanged = (track: TrackType) => setDimensions({
```

Comment:
Do we still need the `?` postfix operator on `track` here? We could also do something like `setDimensions({ ...track?.dimensions })` so we get all of the properties like we used to, just in case something relied on any other `track` properties.

Reply:
If you need the null coaslescing, the types are wrong, so change those rather than reinstating what appears to be a redundant operator.

You're probably right about cloning the track dimensions to keep other properties, although you might have other deeply-nested objects that will suffer from the 'same reference' problem.

It might be better to specify exactly the return shape of this hook, rather than have it infer it from elsewhere.

Reply:
> Do we still need the ? postfix operator on track here?

No - it isn't needed inside the `if (track) {` block.

```
if (track) {
  // track will never be undefined in here
}
```

> We could also do something like setDimensions({ ...track?.dimensions }) so we get all of the properties like we used to, just in case something relied on any other track properties.

We could, but `width` and `height` are the only properties on this object, so I think the code is fine as is. https://media.twiliocdn.com/sdk/js/video/releases/2.10.0/docs/VideoTrack.html#.Dimensions